File: tfl_dsa/python/linklist/linkedist/linkedlist.py
from .node import Node

class LinkedList:
    
    
    def __init__(self):
        self.head =None
        
    def insert(self,item):
        theNode = Node(item)
        
        if self.head ==None:
            self.head =theNode
            return
        
        current=self.head
        while(current.next !=None):
            current = current.next
        
        current.next =theNode
        
        return
    
    
    def search(self,value):
        
        current =self.head
        
        while(current !=None):
            if(current.data==value):
                print(value,"found ")
                return
            
            current =current.next
        print("not found")  
        
        
        return
    
    def remove(self,value):
        
        if self.head == None:
            print("list is empty")
            return
        if self.head.data==value:
            self.head = self.head.next
            return
        
        previousnode =None
        current =self.head
        
        while(current.next!=None):
            previousnode =current
            current=current.next
            if(current.data==value):
                previousnode.next=current.next
                return
            
        print("number not found")
        return
    
    def display(self):
        
        current=self.head
        
        if current==None:
            print("List is empty")
            return
        while (current!=None):
            print(current.data,end= "---->")
            current=current.next
        
        return
    

File: tfl_dsa/java/liskedlist/linkedlist.java
class Node{
    int value;
    Node next;
    public Node(int value){
        this.value =value;
        this.next=null;

    }

}

class operation{
    Node head;

    operation(){
        this.head=null;
    }

    public void insert(int data){
        Node newNode= new Node(data);
        if (head ==null){
            head= newNode;
            return;
        }
        Node current=head;
        while(current.next!= null){
            current=current.next;
        }
        current.next =newNode;
    }

    public void display(){
        if(head== null){
            System.out.println("list is empty");
        }

        Node current = head;
        while(current!= null){
            System.out.print (current.value+"-->");
            current=current.next;
        }
    }
    
    void remove(int data){


    }

    void search(int data){
        
    }
}

class linkedlist{
    public static void main(String[] args) {

        operation op =new operation();
        op.insert(45);
        op.insert(34);
        op.insert(78);
        op.display();
        
    }
}